cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -58,26 +60,62 @@ def __init__(
                 in the form of the t in (-1)**t = exp(i pi t).
 
         Raises:
-            ValueError: If the given pauli string does not have eignevalues +1
-                or -1.
+            ValueError: If coefficient is not 1 or -1.
         """
-        if pauli_string.coefficient == -1:
-            pauli_string = -pauli_string
-            exponent_pos, exponent_neg = exponent_neg, exponent_pos
+        gate = PauliStringPhasorGate(
+            dps.DensePauliString(pauli_string.values(), coefficient=pauli_string.coefficient),
+            exponent_neg=exponent_neg,
+            exponent_pos=exponent_pos,
+        )
+        super().__init__(gate, pauli_string.qubits)
+        self._pauli_gate = gate
```

Comment:
We don't need to store an explicit `self._pauli_gate` because we are already storing the same pauli gate as `self._gate` because of deriving from `GateOperation` and calling `super().__init__`.

Reply:
It was to avoid casting everywhere, but I can change it.

(Ultimately we should probably make GateOperation generic to help in these cases, but outside the scope of this PR).